components/coprocessor_plugin_api/src/util.rs

```diff
@@ -42,8 +47,23 @@ impl BuildInfo {
     }
 }
 
+/// Information about the plugin, like its name and version.
+#[repr(C)]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct PluginInfo {
+    /// The name of the plugin.
+    pub name: &'static str,
+    /// The version string of the plugin. Should follow semantic versioning.
+    pub version: &'static str,
+}
+
 /// Declare a plugin for the library so that it can be loaded by TiKV.
 ///
+/// The macro has three different versions:
+/// * `declare_plugin!(plugin_name, plugin_version, plugin_ctor)` which gives you full control.
```

Comment:
I suggest documenting the type for plugin_name and plugin_version.

Reply:
Ah yes, good idea! Btw: would it be a good idea to require a `literal` matcher instead of an `expr`? https://doc.rust-lang.org/edition-guide/rust-next/literal-macro-matcher.html

Reply:
Good idea

Reply:
done

Reply:
Okay no, not a good idea, because then we can't use `env!()` macro to collect information from Cargo.toml. Going to revert this.

Reply:
😅